pkg/operator/ceph/object/objectstore.go

```diff
@@ -399,47 +399,47 @@ func createMultisite(objContext *Context, endpointArg string) error {
+			return errorOrIsNotFound(getRealmErr, "'radosgw-admin realm get' failed with code %d, for reason %q", strconv.Itoa(code), output)
 		}
 	}
 
 	// create the zonegroup if it doesn't exist yet
-	output, err = RunAdminCommandNoMultisite(objContext, true, "zonegroup", "get", realmArg, zoneGroupArg)
-	if err != nil {
+	output, getZonegroupErr := RunAdminCommandNoMultisite(objContext, true, "zonegroup", "get", realmArg, zoneGroupArg)
+	if getZonegroupErr != nil {
 		// ENOENT means "No such file or directory"
-		if code, err := exec.ExtractExitCode(err); err == nil && code == int(syscall.ENOENT) {
-			output, err = RunAdminCommandNoMultisite(objContext, false, "zonegroup", "create", "--master", realmArg, zoneGroupArg, endpointArg)
-			if err != nil {
-				return errorOrIsNotFound(err, "failed to create ceph zone group %q, for reason %q", objContext.ZoneGroup, output)
+		if code, extractErr := exec.ExtractExitCode(getZonegroupErr); extractErr == nil && code == int(syscall.ENOENT) {
+			output, createZonegroupErr := RunAdminCommandNoMultisite(objContext, false, "zonegroup", "create", "--master", realmArg, zoneGroupArg, endpointArg)
+			if createZonegroupErr != nil {
+				return errorOrIsNotFound(createZonegroupErr, "failed to create ceph zone group %q, for reason %q", objContext.ZoneGroup, output)
 			}
 			logger.Debugf("created zone group %q", objContext.ZoneGroup)
 		} else {
-			return errorOrIsNotFound(err, "'radosgw-admin zonegroup get' failed with code %d, for reason %q", strconv.Itoa(code), output)
+			return errorOrIsNotFound(getZonegroupErr, "'radosgw-admin zonegroup get' failed with code %d, for reason %q", strconv.Itoa(code), output)
 		}
 	}
 
 	// create the zone if it doesn't exist yet
-	output, err = runAdminCommand(objContext, true, "zone", "get")
-	if err != nil {
+	output, getZoneErr := runAdminCommand(objContext, true, "zone", "get")
+	if getZoneErr != nil {
 		// ENOENT means "No such file or directory"
-		if code, err := exec.ExtractExitCode(err); err == nil && code == int(syscall.ENOENT) {
-			output, err = runAdminCommand(objContext, false, "zone", "create", "--master", endpointArg)
-			if err != nil {
-				return errorOrIsNotFound(err, "failed to create ceph zone %q, for reason %q", objContext.Zone, output)
+		if code, extractErr := exec.ExtractExitCode(getZoneErr); extractErr == nil && code == int(syscall.ENOENT) {
+			output, createZoneErr := runAdminCommand(objContext, false, "zone", "create", "--master", endpointArg)
+			if createZoneErr != nil {
+				return errorOrIsNotFound(createZoneErr, "failed to create ceph zone %q, for reason %q", objContext.Zone, output)
 			}
 			logger.Debugf("created zone %q", objContext.Zone)
 		} else {
-			return errorOrIsNotFound(err, "'radosgw-admin zone get' failed with code %d, for reason %q", strconv.Itoa(code), output)
+			return errorOrIsNotFound(getZoneErr, "'radosgw-admin zone get' failed with code %d, for reason %q", strconv.Itoa(code), output)
```

Comment:
The `err` variable would also be perfectly functional with this restructuring.
```go
code, err := exec.ExtractExitCode(err); err == nil && code == int(syscall.ENOENT) 
if {
  ...
} else {
  return errorOrIsNotFound(err, "'radosgw-admin zone get' failed with code %d, for reason %q", strconv.Itoa(code), output)
}
```

Reply:
Not sure If I understood, I will try to discuss in the call

Reply:
After discussing, now it's clear that the `getRealmErr` would be used on this line. The `extractErr` is not relevant since the point of line 435 is only to check if the zone exists. If the error is not ENOENT, then we would return the original error from the `get`. 

We discussed keeping other errors named `err` except where needed to be renamed `getZoneErr` since that is needed to pass down to this scope.